### PR TITLE
Hera/add navigation

### PR DESCRIPTION
**Pull request by MattBudz**
### Summary

This PR adds the navigation to the Hera layout

### Check List

~- [ ] Added a CHANGELOG entry~

### CODE REVIEW

File: app/assets/stylesheets/hera/modules/navigation/_navigation.scss [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+            padding: 0.5rem 0;
+          }
+        }
+      }
+    }
+  }
+
+  @media (max-width: 991px) {
+    overflow: scroll;
+    height: 90vh;
```
Comment (MattBudz) [inline app/assets/stylesheets/hera/modules/navigation/_navigation.scss:378]:
```suggestion
    overflow: auto;
    height: 100vh;
```

Let's not force unnecessary scroll bars when they aren't needed. Also using 100vh allows the menu to cover the full screen and hide that we are scrolling the menu in Pro (vs the view).

File: app/views/shared/navigation/_complementary_nav.html.erb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<ul class="navbar-nav navbar-nav-scroll complementary-nav pt-0">
+  <% if @project %>
+    <li class="nav-item">
+      <%= render "layouts/hera/navbar/search" %>
+    </li>
+  <% end %>
+  <% if defined?(Dradis::Pro) %>
+    <% if (can?(:read, Project) || can?(:read, Team)) && @project.blank? %>
+      <%= content_tag :li, class: class_names('nav-item', active: controller_path == 'trash') do %>
+        <%= link_to main_app.trash_index_path, class: 'nav-link' do %>
```
Comment (caitmich) [inline app/views/shared/navigation/_complementary_nav.html.erb:10]: why is `main_app` needed here?

File: app/views/shared/navigation/_complementary_nav.html.erb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<ul class="navbar-nav navbar-nav-scroll complementary-nav pt-0">
+  <% if @project %>
+    <li class="nav-item">
+      <%= render "layouts/hera/navbar/search" %>
+    </li>
+  <% end %>
+  <% if defined?(Dradis::Pro) %>
+    <% if (can?(:read, Project) || can?(:read, Team)) && @project.blank? %>
```
Comment (caitmich) [inline app/views/shared/navigation/_complementary_nav.html.erb:8]:
```suggestion
    <% if (can?(:read, Project) || can?(:read, Team)) && !@project %>
```

File: app/views/shared/navigation/_complementary_nav.html.erb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      <%= content_tag :li, class: class_names('nav-item', active: controller_path == 'trash') do %>
+        <%= link_to main_app.trash_index_path, class: 'nav-link' do %>
+          <i class="fa-solid fa-trash-can d-none d-lg-inline <%= class_names('d-inline': current_user.role?(:admin)) %>"></i>
+          <span class="d-inline d-lg-none">Bin</span>
```
Comment (caitmich) [inline app/views/shared/navigation/_complementary_nav.html.erb:12]:
```suggestion
          <span class="d-inline d-lg-none">Trash</span>
```

File: app/views/shared/navigation/_complementary_nav.html.erb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      <% end %>
+    <% end %>
+  <% end %>
+  <li class="nav-item dropdown">
+    <a href="javascript:void(0)" class="nav-link dropdown-toggle no-caret-xl" id="configurationDropdown" title="Configuration" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+      <i class="fa-solid fa-cog d-none d-lg-inline" title="Configuration"></i>
+      <span class="d-inline d-lg-none">Configuration</span>
+    </a>
+    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="configurationDropdown">
+      <li><%= link_to 'Configure plugins', main_app.project_configurations_path(current_project), class: 'dropdown-item' %></li>
```
Comment (caitmich) [inline app/views/shared/navigation/_complementary_nav.html.erb:23]: I don't believe this view is functional currently

File: app/views/shared/navigation/_complementary_nav.html.erb (hunk 2 above)
Comment (caitmich) [reply app/views/shared/navigation/_complementary_nav.html.erb:10]: Ditto later in this file. Is there a reason we need `main_app`?

File: app/views/shared/navigation/_complementary_nav.html.erb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      <li><%= link_to "Manage Tags", project_tags_path(current_project), class: 'dropdown-item' %></li>
+    </ul>
+  </li>
+  <li class="nav-item notifications dropdown">
+    <%= 
+      link_to main_app.project_notifications_path(current_project), 
+        class: 'nav-link dropdown-toggle no-caret-xl', 
+        id: 'notificationsDropdown',
```
Comment (caitmich) [inline app/views/shared/navigation/_complementary_nav.html.erb:31]: is this id needed? It looks like we're using the data-attribute to grab this

Comment (caitmich) [reply app/views/shared/navigation/_complementary_nav.html.erb:31]: If so can we rename it to use kebab case?

File: app/views/shared/navigation/_complementary_nav.html.erb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      <i class="notifications-bell fa-solid fa-bell d-none d-lg-inline" title="Notifications"><i class="notifications-dot d-none" data-behavior="notifications-dot"></i></i>
+      <span class="d-inline d-lg-none">Notifications</span>
+    <% end %>
+    <div class="dropdown-menu dropdown-menu-end" data-url="<%= main_app.notifications_path %>"  aria-labelledby="notificationsDropdown">
+    </div>
+  </li>
+  <%= render partial: 'shared/navigation/help' %>
+
+  <li class="nav-item dropdown me-0">
+    <a class="nav-link dropdown-toggle pe-0 py-lg-0 d-flex align-items-center no-caret-xl" href="javascript:void(0)" id="profileDropdown" role="button" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```
Comment (caitmich) [inline app/views/shared/navigation/_complementary_nav.html.erb:48]: Same as above with this id

File: app/views/shared/navigation/_help.html.erb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      <a href="https://discuss.dradis.com?utm_source=ce&utm_medium=app&utm_campaign=forum" target="_blank" class="dropdown-item">Community forums</a>
+    </li>
+    <li>
+      <div class="dropdown-divider"></div>
+    </li>
+    <li>
+      <a href="https://dradis.com/careers/?utm_source=ce&utm_medium=app&utm_campaign=careers" target="_blank" class="dropdown-item">We're hiring</a>
+    </li>
+
+    <% unless defined?(Dradis::Pro) %>
```
Comment (caitmich) [inline app/views/shared/navigation/_help.html.erb:53]: for consistency why not use `!defined?`

File: app/views/shared/navigation/_instance_nav.html.erb [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    <% end %>
+
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target=".dual-nav" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
+      <i class="fa-solid fa-bars"></i>
+    </button>
+
+    <div class="collapse navbar-collapse justify-content-end dual-nav" id="instance_navbar" data-behavior="navbar-collapse">
+      <ul class="navbar-nav navbar-nav-scroll me-auto pb-0">
+        <% if defined?(Dradis::Pro) %>
+          <%= content_tag :li, class:'nav-item' << (controller_path == 'dashboard' ? ' active' : '') do %>
```
Comment (caitmich) [inline app/views/shared/navigation/_instance_nav.html.erb:20]: what about using `class_names(` like we have in project_nav?

File: app/views/shared/navigation/_complementary_nav.html.erb (hunk 2 above)
Comment (MattBudz) [reply app/views/shared/navigation/_complementary_nav.html.erb:10]: This was carried over from Tylium and yes IIRC it's for integrations

File: app/views/shared/navigation/_complementary_nav.html.erb (hunk 6 above)
Comment (nicolachr) [reply app/views/shared/navigation/_complementary_nav.html.erb:31]: Removed as it was not being used

**Pull request merged**